ApollosProject/apollos-prototype
# Don't login a user if their phone number has multiple users

## DESCRIPTION

### What does this PR do, or why is it needed?

New SMS Auth flow: 

1. On login, if multiple people have the same phone number, check to see if any of those people already have a User Account that is the phone number.
2. If Yes, that is the person.
3. If no, create a new person record with the User Account that is the phone number(it would be a duplicate that would have to be merged at some point)

### What design trade-offs/decisions were made?

Opted to avoid trying to guess which phone number is the correct phone number.

### How do I test this PR?

- [x] 1. Spin up this PR (make sure you yarn)
- [x] 2. Make sure you have multiple users with your phone number in Rock, login via SMS
  - You should login as a new user
- [x] 3. Logout of that account, and then login via SMS again 
  - You should login as that user. 
- [x] 4. Using the Rock admin interface, give yourself a new phone number (borrow a friend who isn't in Rock?)  Login via SMS
  - You should login as yourself

### What automated tests did you write?

API Tests covering all new functionality.

## TODO:

- [x] Closes #746 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
